### PR TITLE
Adjust Invite User Flow

### PR DESCRIPTION
**Pull request by lwjameson**
## In this PR

With the switch over to using the Supabase PKCE Auth flow, some adjustments had to be made. 

- The PKCE flow does not support invite user, so in order to keep the same UX the flow needed to be changed.
  - Invite User now uses an API endpoint which creates the user and sends them an email instead of using the Auth inviteUserByEmail functionality.
  - This endpoint creates an encrypted token which includes the user ID as well as the epoch time that it was generated.
  - The user, when clicking the 'Accept Invite' button is sent to a new client page.
  - This page is similar to the reset password page, but sends it's response to a new API endpoint which validates the token and sets their password.

### CODE REVIEW

File: src/apps/accept-org-invite/AcceptOrgInvite.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        } else {
+          setSuccess(true);
+        }
+        setBusy(false);
+      });
+    }
+  };
+
+  return (
+    <div className='reset-password'>
```
Comment (rsimon) [inline src/apps/accept-org-invite/AcceptOrgInvite.tsx:66]: That should probably be changed to something like `accept-org-invite`.

File: template.html [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+              border="0"
+              cellpadding="0"
+              cellspacing="0"
+              class="main"
+            >
+              <!-- START MAIN CONTENT AREA -->
+              <tr>
+                <td class="wrapper">
+                  <img src="{{ .SiteURL }}/img/branding/email/top-logo.svg" />
+                  <p>Hello from Cove Studio</p>
```
Comment (rsimon) [inline template.html:327]: Is this a Cove-specific template? Or something that also needs branding?

File: src/pages/api/accept-new-user.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
+import { createClient } from '@supabase/supabase-js';
+import type { APIRoute } from 'astro';
+import { decrypt } from '@backend/crypto';
+import type { ApiAcceptOrgInvite } from '../../Types';
+import { error } from 'node_modules/astro/dist/core/logger/core';
```
Comment (rsimon) [inline src/pages/api/accept-new-user.ts:5]: That's probably a leftover from a wrong autocomplete?

File: src/pages/api/invite-user.ts [hunk 4]
```diff
@@ -66,13 +85,157 @@ export const POST: APIRoute = async ({ request, cookies }) => {
...
+  const mailOptions = {
+    from: MAIL_FROM_ADDRESS,
+    to: body.email.toLowerCase(),
+    subject: t['Join Recogito'],
+    html: html,
+  };
+
+  const sendResp = await transporter.sendMail(mailOptions);
+
+  console.log('Message sent: ', sendResp.messageId);
```
Comment (rsimon) [inline src/pages/api/invite-user.ts:225]: Leftover debug log :-)

File: src/pages/api/invite-user.ts [hunk 5]
```diff
@@ -66,13 +85,157 @@ export const POST: APIRoute = async ({ request, cookies }) => {
...
+          fontWeight: 'normal',
+          padding: {
+            top: 0,
+            bottom: 16,
+            right: 24,
+            left: 24,
+          },
+        },
+        props: {
+          text: t['_join_recogito_message_'].replaceAll(
```
Comment (rsimon) [inline src/pages/api/invite-user.ts:174]: Changed that to `replaceAll` since I ended up with a DE version that uses the instance name twice.

**Comment by rsimon** [comment]
Nice!

**Review by rsimon — APPROVED** [review]
Nice!

File: template.html (hunk 2 above)
Comment (lwjameson) [reply template.html:327]: This is some helper files.  I will be moving them.

**Pull request merged**